app/assets/javascripts/helpers/discussion/post_helpers.js

```diff
@@ -0,0 +1,38 @@
+//= require helpers/discussion/edit_post
+//= require helpers/discussion/delete_post
+
+var DISCUSSION_POST_HELPERS = (function($, EDIT_DISCUSSION_POST,
+                                           DELETE_DISCUSSION_POST) {
+  /* global JST, Routes */
+  'use strict';
+
+  /**
+   * Shows the comments toolbar for the discussion post.
+   *
+   * @param element
+   */
+  function showCommentToolbar(element, selector) {
+    var $comments = $('.discussion_post', element).filter(selector + '*');
+    $comments.find('.toolbar').show();
+  }
+
+  /**
+   * Initializes the toolbar so that the discussion post can be edited and deleted.
+   *
+   * @param {HTMLElement} element The base element to attach the event to.
+   * @param {String} selector The selector for the specific discussion posts.
+   */
+  function initializeToolbar(element, selector) {
+    showCommentToolbar(element, selector);
+    $(document).on('DOMNodeInserted', function(e) {
```

Comment:
I think it should be judicious. I used it for application-wide components because they are supposed to be global. For specific components, other handlers might be more appropriate.